lib/libspl/include/sys/time.h

```diff
@@ -58,6 +58,12 @@
 #define	NSEC2MSEC(n)    ((n) / (NANOSEC / MILLISEC))
 #endif
 
+#ifndef HAVE_HR_TYPES
```

Comment:
It's not clear to me why this is needed.  It looks to me like `hrtime_t` and `timestruc_t` and `timespec_t` all should have been originally included in `sys/time.h`.  Let's move them where they belong to avoid this conditional code.

```diff
diff --git a/lib/libspl/include/sys/time.h b/lib/libspl/include/sys/time.h
index f0da440..e88435a 100644
--- a/lib/libspl/include/sys/time.h
+++ b/lib/libspl/include/sys/time.h
@@ -58,6 +58,11 @@
 #define        NSEC2MSEC(n)    ((n) / (NANOSEC / MILLISEC))
 #endif
 
+typedef longlong_t      hrtime_t;
+typedef struct timespec timestruc_t;
+typedef struct timespec timespec_t;
+
+
 extern hrtime_t gethrtime(void);
 extern void gethrestime(timestruc_t *);
 
diff --git a/lib/libspl/include/sys/types.h b/lib/libspl/include/sys/types.h
index 25d56be..8996af9 100644
--- a/lib/libspl/include/sys/types.h
+++ b/lib/libspl/include/sys/types.h
@@ -55,10 +55,6 @@ typedef longlong_t   diskaddr_t;
 typedef ulong_t                pgcnt_t;        /* number of pages */
 typedef long           spgcnt_t;       /* signed number of pages */
 
-typedef longlong_t     hrtime_t;
-typedef struct timespec        timestruc_t;
-typedef struct timespec timespec_t;
-
 typedef short          pri_t;
 
 typedef int            zoneid_t;
```